src/control/common/storage/types.go

```diff
@@ -57,64 +57,64 @@ func (nc NvmeControllers) String() string {
 			fmt.Fprintf(&buf, "\t\tNamespace: %+v\n", ns)
 		}
 
-		for _, hs := range ctrlr.Healthstats {
-			fmt.Fprintf(
-				&buf, "\tHealth Stats:\n\t\tTemperature:%dK(%dC)\n",
-				hs.Temp, hs.Temp-273)
-
-			if hs.Tempwarn > 0 {
-				fmt.Fprintf(&buf, "\t\t\tWarning Time:%d\n",
-					uint64(hs.Tempwarn))
-			}
-			if hs.Tempcrit > 0 {
-				fmt.Fprintf(&buf, "\t\t\tCritical Time:%d\n",
-					uint64(hs.Tempcrit))
-			}
-
-			fmt.Fprintf(&buf, "\t\tController Busy Time:%d minutes\n",
-				uint64(hs.Ctrlbusy))
-			fmt.Fprintf(&buf, "\t\tPower Cycles:%d\n",
-				uint64(hs.Powercycles))
-			fmt.Fprintf(&buf, "\t\tPower On Hours:%d hours\n",
-				uint64(hs.Poweronhours))
-			fmt.Fprintf(&buf, "\t\tUnsafe Shutdowns:%d\n",
-				uint64(hs.Unsafeshutdowns))
-			fmt.Fprintf(&buf, "\t\tMedia Errors:%d\n",
-				uint64(hs.Mediaerrors))
-			fmt.Fprintf(&buf, "\t\tError Log Entries:%d\n",
-				uint64(hs.Errorlogs))
-
-			fmt.Fprintf(&buf, "\t\tCritical Warnings:\n")
-			fmt.Fprintf(&buf, "\t\t\tTemperature: ")
-			if hs.Tempwarning {
-				fmt.Fprintf(&buf, "WARNING\n")
-			} else {
-				fmt.Fprintf(&buf, "OK\n")
-			}
-			fmt.Fprintf(&buf, "\t\t\tAvailable Spare: ")
-			if hs.Availspare {
-				fmt.Fprintf(&buf, "WARNING\n")
-			} else {
-				fmt.Fprintf(&buf, "OK\n")
-			}
-			fmt.Fprintf(&buf, "\t\t\tDevice Reliability: ")
-			if hs.Reliability {
-				fmt.Fprintf(&buf, "WARNING\n")
-			} else {
-				fmt.Fprintf(&buf, "OK\n")
-			}
-			fmt.Fprintf(&buf, "\t\t\tRead Only: ")
-			if hs.Readonly {
-				fmt.Fprintf(&buf, "WARNING\n")
-			} else {
-				fmt.Fprintf(&buf, "OK\n")
-			}
-			fmt.Fprintf(&buf, "\t\t\tVolatile Memory Backup: ")
-			if hs.Volatilemem {
-				fmt.Fprintf(&buf, "WARNING\n")
-			} else {
-				fmt.Fprintf(&buf, "OK\n")
-			}
+		hs := ctrlr.Healthstats
+
+		fmt.Fprintf(
+			&buf, "\tHealth Stats:\n\t\tTemperature:%dK(%dC)\n",
+			hs.Temp, hs.Temp-273)
+
+		if hs.Tempwarn > 0 {
+			fmt.Fprintf(&buf, "\t\t\tWarning Time:%d\n",
+				uint64(hs.Tempwarn))
```

Comment:
What does this value represent, exactly? Is it the time that the warning was raised? Is it a duration?

In either case, it would be a lot more useful to the admin as a parsed value displayed in human-comprehensible format. I know that you're just fixing the underlying issue of matching stats to specific devices, but fixing this would a quick win for the user.

Reply:
maybe ask @sydneyvanda and address in another PR

Reply:
@mjmac @tanabarr this is a warning time duration, and yes this can all be cleaned up along with all of the other stats once more discussion about what the health stats output should look like and what stats should be included. I didn't get much input before landing the original patch to include all the stats, wanted to get additional input later.